clinica/iotools/converters/adni_to_bids/adni_to_bids.py

```diff
@@ -63,7 +63,7 @@ def convert_clinical_data(self, clinical_data_dir, out_path):
         bids_subjs_paths = bids.get_bids_subjs_paths(out_path)
         conversion_path = path.join(out_path, "conversion_info")
 
-        if not bids_ids:
+        if clinical_data_only:
```

Comment:
This is where it gets funny. The previous code was supposed to retrieve the `bids_ids` from finding `sub-*` folders. But because `conversion_info` was wrongly included until this fix, this part of the code was never triggered. Now that fetching the list of `bids_id`works properly, this test can actually pass if a previous processing of the imaging data did not yield any conversion (such as with the dataset of the bug report).

Reply:
If the test pass then all participant IDs are fetched from the clinical data and you end up with a final conversion a full `participants.tsv` and no `sub-*`imaging folders.

Reply:
To sum-up, this change distinguish the following two use cases:

- The user specifically asks for all clinical data with `clinical_data_only`
- The user did not ask for it and the imaging data conversion produced no data

Reply:
Clarified in a call